Failure on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.43014&attempt_id=3ddc6780-1600-48bd-a6ff-ddfab48aa474)

Attempted: wcc

<details><summary>Partial log (click to expand)</summary><p>

```
make[2]: Entering directory '/build/source/src/wcc'
gcc -W -Wall -Wno-discarded-qualifiers -Wno-int-conversion -Wno-unused-parameter -Wno-unused-function -Wno-unused-result -fpie -pie -fPIC -g3 -ggdb -I../../include  -I./include/sflib/ -I./include -I../../include/  -Wno-incompatible-pointer-types  -fstack-protector-all -Wl,-z,relro,-z,now -DPACKAGE -DPACKAGE_VERSION -masm=intel -rdynamic -D_FORTIFY_SOURCE=2 -O2 wcc.c -o wcc -lbfd -lelf -lcapstone
gcc: error: unrecognized command line option '-masm=intel'
make[2]: *** [Makefile:29: all] Error 1
make[2]: Leaving directory '/build/source/src/wcc'
make[1]: *** [Makefile:11: all] Error 2
make[1]: Leaving directory '/build/source/src'
make: *** [Makefile:14: all] Error 2
builder for '/nix/store/kqwsihkdmc2j3jshaw2si3myljnhh0na-wcc-unstable-2018-04-05.drv' failed with exit code 2
error: build of '/nix/store/kqwsihkdmc2j3jshaw2si3myljnhh0na-wcc-unstable-2018-04-05.drv' failed
```
</p></details>

--- a/pkgs/development/compilers/wcc/default.nix
+++ b/pkgs/development/compilers/wcc/default.nix
@@ -1,0 +1,49 @@
+{ stdenv, fetchFromGitHub, capstone, libbfd, libelf, libiberty, readline }:
+
+stdenv.mkDerivation rec {
+  name = "wcc-unstable-${version}";
+  version = "2018-04-05";
+
+  src = fetchFromGitHub {
+    owner = "endrazine";
+    repo = "wcc";
+    rev = "f141963ff193d7e1931d41acde36d20d7221e74f";
+    sha256 = "1f0w869x0176n5nsq7m70r344gv5qvfmk7b58syc0jls8ghmjvb4";
+    fetchSubmodules = true;
+  };
+
+  buildInputs = [ capstone libbfd libelf libiberty readline ];
+
+  postPatch = ''
+    sed -i src/wsh/include/libwitch/wsh.h src/wsh/scripts/INDEX \
+      -e "s#/usr/share/wcc#$out/share/wcc#"
+  '';
+
+  installFlags = [ "DESTDIR=$(out)" ];
+
+  preInstall = ''
+    mkdir -p $out/usr/bin
+  '';
+
+  postInstall = ''
+    mv $out/usr/* $out
+    rmdir $out/usr
+    mkdir -p $out/share/man/man1
+    cp doc/manpages/*.1 $out/share/man/man1/
+  '';
+
+  preFixup = ''
+    # Let patchShebangs rewrite shebangs with wsh.
+    PATH+=:$out/bin
+  '';
+
+  enableParallelBuilding = true;
+
+  meta = with stdenv.lib; {
+    homepage = https://github.com/endrazine/wcc;
+    description = "Witchcraft compiler collection: tools to convert and script ELF files";
+    license = licenses.mit;
+    platforms = [ "x86_64-linux" ];
+    maintainers = with maintainers; [ orivej ];
+  };
+}

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -7113,6 +7113,8 @@ with pkgs;
 
   valadoc = callPackage ../development/tools/valadoc { };
 
+  wcc = callPackage ../development/compilers/wcc { };
+
   wla-dx = callPackage ../development/compilers/wla-dx { };
 
   wrapCCWith =

